plugins/org.eclipse.reddeer.junit/src/org/eclipse/reddeer/junit/requirement/matcher/RequirementVersionMatcher.java

```diff
@@ -34,22 +34,17 @@
  * @author mlabuda@redhat.com
  *
  */
-public class RequirementVersionMatcher extends AbstractRequirementMatcher {
+public class RequirementVersionMatcher extends RequirementAttributeMatcher {
+	
+	private String matchingValue;
```

Comment:
You are correct, i cannot change attributeName to Matcher. But i did a few changes and IMHO now its a bit better.
New way of restricting the requirements is:

`new RequirementMatcher(CustomJavaAnnotation.class, "version" , new VersionMatcher("1.8;1.9]"))`

or

`new RequirementMatcher(CustomJavaAnnotation.class, "name" , new RegexMatcher("some_regex"))`


or

`new RequirementMatcher(CustomJavaAnnotation.class, "name" , "some_value")`
which internally translates to IsEqual matcher. This is approach is the same as we do in other places in reddeer :)

notice that I removed AttributeRegexMatcher since simple RegexMatcher can be used. I also renamed AttributeVersionMatcher to VersionMatcher and moved it to common plugin as it can match any string (not tied to Requirements)